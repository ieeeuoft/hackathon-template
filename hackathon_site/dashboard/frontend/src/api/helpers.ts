import {
    Order,
    OrderInTable,
    OrderItemTableRow,
    ReturnedItem,
    ReturnOrderInTable,
} from "api/types";

export const formatDateTime = (dateTimeString: string): string => {
    const dateTime = new Date(dateTimeString);
    const options: Intl.DateTimeFormatOptions = {
        month: "long",
        day: "numeric",
        year: "numeric",
        hour: "numeric",
        minute: "numeric",
        hour12: true,
    };
    return dateTime.toLocaleString("en-US", options);
};
export const teamOrderListSerialization = (
    orders: Order[]
): {
    pendingOrders: OrderInTable[];
    checkedOutOrders: OrderInTable[];
    returnedOrders: ReturnOrderInTable[];
    hardwareIdsToFetch: number[];
} => {
    const pendingOrders: OrderInTable[] = [];
    const checkedOutOrders: OrderInTable[] = [];
    const returnedOrders: ReturnOrderInTable[] = [];
    const hardwareIdsToFetch: Record<number, number> = {};
    orders.forEach((order) => {
        if (order.status !== "Cancelled") {
            const hardwareItems: Record<number, OrderItemTableRow> = {};
            const hardwareRequested: Record<number, number> = {};
            const returnedItems: Record<string, ReturnedItem> = {};
            order.request.forEach(
                (hardware) =>
                    (hardwareRequested[hardware.id] = hardware.requested_quantity)
            );
            order.items.forEach(({ id, hardware_id, part_returned_health }) => {
                if (part_returned_health) {
                    const returnItemKey = `${hardware_id}-${part_returned_health}`;
                    if (returnedItems[returnItemKey])
                        returnedItems[returnItemKey].quantity += 1;
                    else {
                        const date = new Date(order.updated_at);
                        returnedItems[returnItemKey] = {
                            id,
                            quantity: 1,
                            part_returned_health,
                            hardware_id,
                            time: `${date.toLocaleTimeString()} (${date.toDateString()})`,
                        };
                    }
                } else {
                    if (hardwareItems[hardware_id])
                        hardwareItems[hardware_id].quantityGranted += 1;
                    else
                        hardwareItems[hardware_id] = {
                            id: hardware_id,
                            quantityGranted: 1,
                            quantityRequested: hardwareRequested[hardware_id],
                        };
                }
                hardwareIdsToFetch[hardware_id] = hardware_id;
            });
            const returnedHardware = Object.values(returnedItems);
            if (returnedHardware.length)
                returnedOrders.push({
                    id: order.id,
                    hardwareInOrder: returnedHardware,
                });

            const hardwareInTableRow = Object.values(hardwareItems);
            if (hardwareInTableRow.length > 0)
                (order.status === "Submitted" || order.status === "Ready for Pickup"
                    ? pendingOrders
                    : checkedOutOrders
                ).push({
                    id: order.id,
                    status: order.status,
                    hardwareInTableRow,
                    createdTime: order.created_at,
                    updatedTime: order.updated_at,
                });
        }
    });
    return {
        pendingOrders: [...pendingOrders].sort((a, b) => b.id - a.id),
        checkedOutOrders: [...checkedOutOrders].sort((a, b) => b.id - a.id),
        returnedOrders,
        hardwareIdsToFetch: Object.values(hardwareIdsToFetch),
    };
};

export const sortReturnedOrders = (
    order1: ReturnOrderInTable,
    order2: ReturnOrderInTable
): number => {
    const orderDate1 = order1.hardwareInOrder[0].time;
    const orderDate2 = order2.hardwareInOrder[0].time;

    const matchResult = orderDate1.match(
        /(\d{1,2}):(\d{2}):(\d{2}) (AM|PM) \((\w{3}) (\w{3}) (\d{2}) (\d{4})\)/
    );
    const matchResult2 = orderDate2.match(
        /(\d{1,2}):(\d{2}):(\d{2}) (AM|PM) \((\w{3}) (\w{3}) (\d{2}) (\d{4})\)/
    );

    if (matchResult && matchResult2) {
        // converting invalid date to a valid Date for first order to be compared
        const [, hours, minutes, seconds, meridiem, , month, day, year] = matchResult;
        const monthIndex = [
            "Jan",
            "Feb",
            "Mar",
            "Apr",
            "May",
            "Jun",
            "Jul",
            "Aug",
            "Sep",
            "Oct",
            "Nov",
            "Dec",
        ].indexOf(month);

        const date = new Date(
            Number(year),
            monthIndex,
            Number(day),
            Number(hours) + (meridiem === "PM" ? 12 : 0),
            Number(minutes),
            Number(seconds)
        );
        const formattedDate = date.toISOString();

        // converting invalid date to a valid Date for first order to be compared
        const [, hours2, minutes2, seconds2, meridiem2, , month2, day2, year2] =
            matchResult2;
        const monthIndex2 = [
            "Jan",
            "Feb",
            "Mar",
            "Apr",
            "May",
            "Jun",
            "Jul",
            "Aug",
            "Sep",
            "Oct",
            "Nov",
            "Dec",
        ].indexOf(month2);

        const date2 = new Date(
            Number(year2),
            monthIndex2,
            Number(day2),
            Number(hours2) + (meridiem2 === "PM" ? 12 : 0),
            Number(minutes2),
            Number(seconds2)
        );
        const formattedDate2 = date2.toISOString();

        return new Date(formattedDate2).valueOf() - new Date(formattedDate).valueOf();
    }
    return 0;
};

export const sortCheckedOutOrders = (
    order1: OrderInTable,
    order2: OrderInTable
): number => {
    return (
        new Date(order2.updatedTime).valueOf() - new Date(order1.updatedTime).valueOf()
    );
};

export const sortPendingOrders = (orders: OrderInTable[]): OrderInTable[] => {
    let ready_orders = [];
    let submitted_orders = [];
    for (let order of orders) {
        if (order.status === "Ready for Pickup") {
            ready_orders.push(order);
        } else {
            submitted_orders.push(order);
        }
    }
    ready_orders.sort((order1, order2) => {
        return (
            new Date(order1.updatedTime).valueOf() -
            new Date(order2.updatedTime).valueOf()
        );
    });

    submitted_orders.sort((order1, order2) => {
        return (
            new Date(order1.updatedTime).valueOf() -
            new Date(order2.updatedTime).valueOf()
        );
    });

    orders.splice(0, orders.length, ...submitted_orders, ...ready_orders);
    return orders;
};
