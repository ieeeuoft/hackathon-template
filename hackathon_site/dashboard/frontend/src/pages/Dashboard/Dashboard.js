import React from "react";
import styles from "./Dashboard.module.scss";
import ConnectedSponsorCard from "components/dashboard/SponsorCard/SponsorCard";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DashCard from "components/dashboard/DashCard/DashCard";
import OpenInNew from "@material-ui/icons/OpenInNew";
import GetApp from "@material-ui/icons/GetApp";

let testTitle = "Hello";
let testContent = [
    { name: "Test", url: "www.facebook.com", icon: <GetApp /> },
    { name: "Test", url: "www.facebook.com", icon: <OpenInNew /> },
];

const Dashboard = () => {
    return (
        <div className={styles.dashboard}>
            <Typography variant="h1">Hackathon Name Hardware Dashboard</Typography>
            <Grid
                container
                direction="row"
                justify="flex-start"
                alignItems="flex-start"
                spacing={2}
                className={styles.dashboardGrid}
            >
                <DashCard title={testTitle} content={testContent} />
                <DashCard title={testTitle} content={testContent} />
                <ConnectedSponsorCard />
            </Grid>
        </div>
    );
};

export default Dashboard;
