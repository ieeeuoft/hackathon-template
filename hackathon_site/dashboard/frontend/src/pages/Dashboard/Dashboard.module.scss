.dashboard {
    width: 100%;

    &Grid {
        margin-top: 30px;
        flex-wrap: nowrap;
    }
}
