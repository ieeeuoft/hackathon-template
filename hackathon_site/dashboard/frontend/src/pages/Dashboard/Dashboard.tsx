import React, { useEffect } from "react";
import styles from "./Dashboard.module.scss";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import DashCard from "components/dashboard/DashCard/DashCard";
import TeamCard from "components/dashboard/TeamCard/TeamCard";
import {
    PendingTables,
    CheckedOutTables,
    ReturnedTable,
} from "components/dashboard/ItemTable/ItemTable";
import ProductOverview from "components/inventory/ProductOverview/ProductOverview";
import Header from "components/general/Header/Header";
import { cardItems } from "testing/mockData";
import { hackathonName } from "constants.js";
import { useDispatch, useSelector } from "react-redux";
import {
    getCurrentTeam,
    isLoadingSelector,
    teamCodeSelector,
    teamSizeSelector,
} from "slices/event/teamSlice";
import { fulfillmentErrorSelector } from "slices/hardware/cartSlice";
import LinearProgress from "@material-ui/core/LinearProgress";
import {
    getTeamOrders,
    hardwareInOrdersSelector,
    orderErrorSelector,
    isLoadingSelector as areOrdersLoadingSelector,
} from "slices/order/orderSlice";
import { getHardwareWithFilters, setFilters } from "slices/hardware/hardwareSlice";
import { getCategories } from "slices/hardware/categorySlice";
import AlertBox from "components/general/AlertBox/AlertBox";
import { openTeamModalItem } from "slices/ui/uiSlice";
import EditTeam from "components/dashboard/EditTeam/EditTeam";
import DateRestrictionAlert from "components/general/DateRestrictionAlert/DateRestrictionAlert";

const Dashboard = () => {
    const dispatch = useDispatch();
    const isTeamLoading = useSelector(isLoadingSelector);
    const areOrdersLoading = useSelector(areOrdersLoadingSelector);
    const orderFulfillmentError = useSelector(fulfillmentErrorSelector);
    const fetchOrderError = useSelector(orderErrorSelector);
    const hardwareInOrders = useSelector(hardwareInOrdersSelector);
    const team_code = useSelector(teamCodeSelector);
    const team_size = useSelector(teamSizeSelector);

    useEffect(() => {
        dispatch(getCurrentTeam());
        dispatch(getCategories());
        dispatch(getTeamOrders());
    }, [dispatch]);

    useEffect(() => {
        if (hardwareInOrders) {
            dispatch(setFilters({ hardware_ids: hardwareInOrders }));
            dispatch(getHardwareWithFilters());
        }
    }, [dispatch, hardwareInOrders]);

    const openEditTeamModal = () => dispatch(openTeamModalItem());

    return (
        <>
            <Header />
            <ProductOverview showAddToCartButton />
            <EditTeam
                teamCode={team_code == null ? "None" : team_code}
                canChangeTeam={true}
                teamSize={team_size}
            />
            <div className={styles.dashboard}>
                <Typography variant="h1">{hackathonName} Hardware Dashboard</Typography>
                <DateRestrictionAlert />
                {isTeamLoading || areOrdersLoading ? (
                    <LinearProgress
                        style={{ width: "100%", marginTop: 25 }}
                        data-testid="team-linear-progress"
                    />
                ) : (
                    <>
                        <Grid
                            container
                            direction="row"
                            justifyContent="flex-start"
                            alignItems="flex-start"
                            spacing={2}
                            className={styles.dashboardGrid}
                        >
                            <Grid
                                item
                                md={3}
                                sm={4}
                                xs={6}
                                className={styles.dashboardGridItem}
                                key={0}
                                data-testid="team"
                            >
                                <TeamCard handleEditTeam={openEditTeamModal} />
                            </Grid>
                            {cardItems.map(({ title, content }, i) => (
                                <Grid
                                    item
                                    md={3}
                                    sm={4}
                                    xs={6}
                                    className={styles.dashboardGridItem}
                                    key={i + 1}
                                >
                                    <DashCard title={title} content={content} />
                                </Grid>
                            ))}
                        </Grid>
                        {orderFulfillmentError && (
                            <AlertBox
                                error={orderFulfillmentError.errors.map(
                                    ({ message }) => message
                                )}
                                title={`There were modifications made to order ${orderFulfillmentError.order_id}`}
                            />
                        )}
                        {fetchOrderError && <AlertBox error={fetchOrderError} />}
                        {/* TODO: add back in when incident reports are completed on the frontend */}
                        {/* <BrokenTable items={itemsBroken} openReportAlert={openBrokenTable} /> */}
                        <PendingTables />
                        <CheckedOutTables />
                        <ReturnedTable />
                    </>
                )}
            </div>
        </>
    );
};

export default Dashboard;
