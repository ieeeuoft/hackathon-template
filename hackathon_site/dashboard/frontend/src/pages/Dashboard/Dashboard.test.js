import React from "react";
import { render } from "@testing-library/react";
import Dashboard from "./Dashboard";
import { cardItems } from "testing/mockData";
import { withRouter, withStore } from "testing/helpers";

it("Renders correctly when the dashboard appears 4 cards and 3 tables", () => {
    const { queryByText, getByText } = render(withStore(withRouter(<Dashboard />)));
    for (let e of cardItems) {
        expect(queryByText(e.title)).toBeTruthy();
    }
    expect(getByText("Returned items")).toBeInTheDocument();
    expect(getByText("Checked out items")).toBeInTheDocument();
    expect(queryByText("Orders pending")).toBeInTheDocument();
    expect(queryByText("Reported broken/lost items")).toBeInTheDocument();
});
