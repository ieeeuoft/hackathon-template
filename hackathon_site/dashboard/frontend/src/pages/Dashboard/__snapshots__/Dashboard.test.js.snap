// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when the dashboard appears with the sponsor card 1`] = `
<DocumentFragment>
  <div
    class="dashboard"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1"
    >
      Hackathon Name Hardware Dashboard
    </h1>
    <div
      class="MuiGrid-root dashboardGrid MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-flex-start"
    >
      <div
        class="MuiGrid-root DashCard MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Hello
        </h2>
        <div
          class="MuiPaper-root paper MuiPaper-elevation3"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
            href="www.facebook.com"
            rel="noopener"
            target="_blank"
          >
            <div
              class="MuiContainer-root items MuiContainer-maxWidthLg"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Test
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                />
              </svg>
            </div>
          </a>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
            href="www.facebook.com"
            rel="noopener"
            target="_blank"
          >
            <div
              class="MuiContainer-root items MuiContainer-maxWidthLg"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Test
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                />
              </svg>
            </div>
          </a>
        </div>
      </div>
      <div
        class="MuiGrid-root DashCard MuiGrid-item"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Hello
        </h2>
        <div
          class="MuiPaper-root paper MuiPaper-elevation3"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
            href="www.facebook.com"
            rel="noopener"
            target="_blank"
          >
            <div
              class="MuiContainer-root items MuiContainer-maxWidthLg"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Test
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                />
              </svg>
            </div>
          </a>
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
            href="www.facebook.com"
            rel="noopener"
            target="_blank"
          >
            <div
              class="MuiContainer-root items MuiContainer-maxWidthLg"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Test
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                />
              </svg>
            </div>
          </a>
        </div>
      </div>
      <div
        class="MuiGrid-root sponsors MuiGrid-item"
        data-testid="sponsor-card"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
        >
          Thanks to our sponsors!
        </h2>
        <div
          class="MuiPaper-root sponsorsPaper MuiPaper-elevation3 MuiPaper-rounded"
        >
          <div
            class="slick-slider slick-initialized"
            dir="ltr"
          >
            <button
              class="slick-arrow slick-prev"
              data-role="none"
              style="display: block;"
              type="button"
            >
               Previous
            </button>
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-2"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="FacultyofAppliedScienceandEngineering.png"
                      src="FacultyofAppliedScienceandEngineering.png"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="-1"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="Huawei.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="AMD.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active"
                  data-index="1"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="CityofBrampton.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="2"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="CognitiveSystems.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="3"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="ECE.png"
                      src="ECE.png"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="4"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="ecobee.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="5"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="FacultyofAppliedScienceandEngineering.png"
                      src="FacultyofAppliedScienceandEngineering.png"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide"
                  data-index="6"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="Huawei.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="7"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="AMD.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="8"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="CityofBrampton.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="9"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="CognitiveSystems.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="10"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="ECE.png"
                      src="ECE.png"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="11"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="ecobee.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="12"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="FacultyofAppliedScienceandEngineering.png"
                      src="FacultyofAppliedScienceandEngineering.png"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
                <div
                  aria-hidden="true"
                  class="slick-slide slick-cloned"
                  data-index="13"
                  style="width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <img
                      alt="Huawei.svg"
                      src="[object Object]"
                      style="width: 100%; display: inline-block;"
                      tabindex="-1"
                    />
                  </div>
                </div>
              </div>
            </div>
            <button
              class="slick-arrow slick-next"
              data-role="none"
              style="display: block;"
              type="button"
            >
               Next
            </button>
            <ul
              class="slick-dots"
              style="display: block;"
            >
              <li
                class="slick-active"
              >
                <button>
                  1
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  2
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  3
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  4
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  5
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  6
                </button>
              </li>
              <li
                class=""
              >
                <button>
                  7
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
